directives/animate/animate.js

```diff
@@ -0,0 +1,268 @@
+
+		const id = ++this.id;
+
+		let outerResolve;
+		const onTransitionEnd = () => {
+			this.clone.removeEventListener('transitionend', onTransitionEnd);
+			if (this.id === id) {
+				this.clone.remove();
+				this.clone = null;
+				if (info.elem.styleAttr !== null) {
+					this.elem.setAttribute('style', info.elem.styleAttr);
+				} else {
+					this.elem.removeAttribute('style');
+				}
+				info.onTransitionEnd();
+				this.dispatchEvent();
+				outerResolve();
+			}
+		};
+		this.clone.addEventListener('transitionend', onTransitionEnd);
+
+		Object.assign(this.clone.style, info.clone.start);
+		Object.assign(this.elem.style, info.elem.start);
+
+		if (info.elem.transition) {
+			await new Promise((r) => requestAnimationFrame(r));
+			Object.assign(this.elem.style, info.elem.transition);
+		}
+
+		await new Promise((r) => requestAnimationFrame(r));
+		Object.assign(this.clone.style, info.clone.end);
+		Object.assign(this.elem.style, info.elem.end);
+
+		return new Promise((resolve) => outerResolve = resolve);
+
+	}
+
+	dispatchEvent() {
+		this.elem.dispatchEvent(
+			new CustomEvent(
+				'd2l-animate-complete',
+				{ bubbles: true, composed: false }
+			)
+		);
+	}
+
+	async getElemInfo() {
+
+		await new Promise((r) => requestAnimationFrame(r));
+		const style = window.getComputedStyle(this.elem);
+
+		const styleAttr = this.elem.getAttribute('style');
+		if (style.display !== 'flex') {
+			this.elem.style.display = 'grid'; // forces margins to collapse during size calculation
+		}
+
+		const rect = this.elem.getBoundingClientRect();
+
+		let top = 0;
+		let left = 0;
+		const offsetParent = this.elem.offsetParent;
+		let e = this.elem;
+		while (e && e !== document && !(e instanceof DocumentFragment)) {
+			if (e === offsetParent) {
+				break;
+			}
+			top += e.offsetTop - (e.scrollTop || 0);
+			left += e.offsetLeft - (e.scrollLeft || 0);
+			e = getComposedParent(e);
+		}
+
+		this.elem.style.removeProperty('display');
+
+		const marginsH = (parseInt(style.marginLeft) || 0) + (parseInt(style.marginRight) || 0);
+		const marginsV = (parseInt(style.marginTop) || 0) + (parseInt(style.marginBottom) || 0);
+
+		let cloneHeight = 0;
+		if (this.clone !== null) {
+			const cloneRect = this.clone.getBoundingClientRect();
+			cloneHeight = cloneRect.height;
+		}
+
+		return {
+			clone: {
+				height: cloneHeight,
+				fullHeight: rect.height + marginsV,
+				width: rect.width + marginsH
+			},
+			elem: {
+				height: rect.height,
+				left,
+				opacity: this.state === 'showing' && this.clone === null ? '0' : style.opacity,
+				styleAttr,
+				top,
+				width: rect.width
+			}
+		};
+
+	}
+
+	async hide(opts) {
+
+		if (this.state === 'hiding' || this.state === 'hidden') {
+			return;
+		}
+		this.state = 'hiding';
+
+		// if focus is inside and was placed by keyboard, move it to next focusable
+		const activeElem = getComposedActiveElement();
+		if (activeElem && activeElem.classList.contains('focus-visible')) {
```

Comment:
Slight tweak here to only move focus if focus is visible. This will limit the places that it'll work, but I was finding it weird to click on the "Remove" button and have focus move on me, vs. when using the keyboard having it move was required.